builtin/logical/database/path_roles.go

```diff
@@ -346,6 +353,25 @@ func (b *databaseBackend) pathRoleCreateUpdate() framework.OperationFunc {
 			if err != nil {
 				return nil, err
 			}
+
+			// TODO need to check that we're only updating the revocation statements or rotation frequency
+
+			// In case this is an update, remove any previous version of the item from the queue
+			if _, err := b.credRotationQueue.PopItemByKey(name); err != nil {
+				if _, ok := err.(*queue.ErrItemNotFound); !ok {
+					return nil, err
+				}
+			}
+
+			// Add their rotation to the queue
+			if err := b.credRotationQueue.PushItem(&queue.Item{
+				Key:      name,
```

Comment:
I originally planned on just the `name` here as shown, but now I wonder how/if namespaces could affect this. Could namespaces be an issue?